specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json

```diff
@@ -3500,6 +3556,11 @@
             "modelAsString": true
           },
           "description": "Allow or disallow public network access to Storage Account. Enabled by default."
+        },
+        "immutableStorageWithVersioning": {
+          "$ref": "#/definitions/ImmutableStorageAccount",
+          "x-ms-client-name": "ImmutableStorageWithVersioning",
+          "description": "The property is immutable and can only be set to true at the account creation time. When set to true, it enables object level immutability for all the containers in the account by default."
```

Comment:
Update account level VLW in my test fail, repro step:
1. create account with account level VLW enabled
2. update account to disable it, Get error `{"error":{"code":"PropertyIsImmutable","message":"The property 'immutableStorageWithVersioning.enabled' was specified in the input, but it cannot be updated as it is immutable."}}`, request ID `x-ms-request-id: c8f2df85-9cf8-4561-b48b-fa5f724a49e3`

Is the error expected? Should we update property description to include the limitation?

Reply:
The description already says that the property is immutable and the server behavior is expected

Reply:
If user can't change it in update account, why add it to update account parameters? 
All properties here should be allowed to update in update storage account API.

Reply:
ImmutableStorageWithVersioning is an object only 1 property enabled is immutable, the ImmutableStorageWithVersioning.immutablityPolicy is mutable

Reply:
The description of property "immutableStorageWithVersioning" says "The property is immutable...", so for common understanding, the whole property is immutable. The description need change to make it more clear.

Besides that, as the value of "enabled" should always be true, can't be other value in create/update account, we might should not add it to rest API, to avoid user error.

Reply:
I will update the comment, However, it is absolutely legal to set enabled=false to account creation time.
Also, we need enabled property here because immutablityPolicy is optional the enabled property will enable VLW for all the containers under such an account without any account level immutabilityPolicy. 
The enabled = true will allow customers to exercise blob level immutability apis.

Reply:
> it is absolutely legal to set enabled=false to account creation time.

In my test, create account with set enabled=false with error {"error":{"code":"InvalidStorageAccountImmutabilityPolicy","message":"Storage account level default immutability policy cannot be set as account level WORM is disabled"}}, request ID x-ms-request-id: b5293f9d-3fab-461a-9c9e-fa2b8b5901c8